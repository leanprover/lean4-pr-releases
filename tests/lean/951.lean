inductive ThingA where
| mkA
deriving Ord
instance : LE ThingA where
  le a b := (compare a b).isLE

instance (t₁ t₂ : ThingA) : Decidable (t₁ <= t₂) := inferInstance
-- TODO: we may want to suppress the name of nested instances
#check instDecidableLeThingA

inductive ThingB where
| mkB
deriving Ord
instance : LE ThingB where
  le a b := (compare a b).isLE
instance (t₁ t₂ : ThingB) : Decidable (t₁ <= t₂) := inferInstance
#check instDecidableLeThingB
