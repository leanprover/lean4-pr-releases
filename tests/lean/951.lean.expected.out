instDecidableLeThingA_951 (t₁ t₂ : ThingA) : Decidable (t₁ ≤ t₂)
instDecidableLeThingB_951 (t₁ t₂ : ThingB) : Decidable (t₁ ≤ t₂)
